lib/ecl/ecl_grid.cpp

```diff
@@ -1871,7 +1871,29 @@ static void ecl_grid_set_active_index(ecl_grid_type * ecl_grid) {
   } else {
     /* --- More involved path in the case of coarsening groups. --- */
 
-    /* 1: Go through all the cells and set the active index. In the
+    /* 1: Reset cell active_index across the entire grid.
+          In the involved path ecl_coarse_cell_update_index() only updates
+          the coarse cells' active_index if it is -1. */
+    for (global_index = 0; global_index < ecl_grid->size; global_index++) {
+        ecl_cell_type * cell = ecl_grid_get_cell(ecl_grid, global_index);
+        if (cell->coarse_group == COARSE_GROUP_NONE)
+        {
+            if (cell->active & CELL_ACTIVE_MATRIX)
+            {
+                cell->active_index[MATRIX_INDEX] = -1;
+            }
+
+            if (cell->active & CELL_ACTIVE_FRACTURE)
+            {
+                cell->active_index[FRACTURE_INDEX] = -1;
+            }
+        } else {
+            ecl_coarse_cell_type * coarse_cell = ecl_grid_iget_coarse_group(ecl_grid, cell->coarse_group);
+            ecl_coarse_cell_reset_active_index(coarse_cell, cell->active);           
```

Comment:
If my comment in `ecl_coarse_grid.cpp` is correct, then we do not need to loop through all the cells in order to retrieve the coarse cells. We can directly access the list of coarse cells in the grid and reset them all

Reply:
I've updated the pull request. I don't really have any more time to put into this so I hope this is acceptable.